### PR TITLE
Cache jvm system parameters to prevent each call from being executed, affecting performance

### PR DESCRIPTION
**Pull request by CrazyHZM**
Signed-off-by: crazyhzm <crazyhzm@gmail.com>

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-common/src/main/java/org/apache/dubbo/common/config/SystemConfiguration.java [hunk 1]
```diff
@@ -18,15 +18,27 @@
...
+    private final Map<String, Object> cache = new ConcurrentHashMap<>();
+
     @Override
     public Object getInternalProperty(String key) {
-        return System.getProperty(key);
+        if (cache.containsKey(key)) {
+            return cache.get(key);
+        } else {
+            Object val = System.getProperty(key);
+            cache.putIfAbsent(key, val);
```
Comment (AlbumenJ) [inline dubbo-common/src/main/java/org/apache/dubbo/common/config/SystemConfiguration.java:39]: val may null here

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Review by AlbumenJ — APPROVED** [review]
LGTM

**Comment by CrazyHZM** [comment]
It will cause System.setProperty() to no longer work. The initial idea is to decide whether to cache according to the key, and some keys do not need to support runtime modification.

**Comment by AlbumenJ** [comment]
> It will cause System.setProperty() to no longer work. The initial idea is to decide whether to cache according to the key, and some keys do not need to support runtime modification.

可以提供下性能压测对比，看下差距再决定是否这样修改

**Comment by CrazyHZM** [comment]
优化前：
Benchmark                               Mode      Cnt    Score    Error   Units
Client.createUser                      thrpt        3   21.763 ± 71.882  ops/ms
Client.existUser                       thrpt        3   50.595 ± 36.093  ops/ms
Client.getUser                         thrpt        3   21.819 ± 48.590  ops/ms
Client.listUser                        thrpt        3    2.916 ±  4.760  ops/ms
Client.createUser                       avgt        3    3.456 ± 31.127   ms/op
Client.existUser                        avgt        3    1.653 ± 15.300   ms/op
Client.getUser                          avgt        3    1.484 ±  1.135   ms/op
Client.listUser                         avgt        3   10.568 ± 29.212   ms/op
Client.createUser                     sample   318690    3.010 ±  0.023   ms/op
Client.createUser:createUser·p0.00    sample             0.481            ms/op
Client.createUser:createUser·p0.50    sample             2.236            ms/op
Client.createUser:createUser·p0.90    sample             4.915            ms/op
Client.createUser:createUser·p0.95    sample             6.709            ms/op
Client.createUser:createUser·p0.99    sample            17.039            ms/op
Client.createUser:createUser·p0.999   sample            48.824            ms/op
Client.createUser:createUser·p0.9999  sample           170.918            ms/op
Client.createUser:createUser·p1.00    sample           178.782            ms/op
Client.existUser                      sample  1547178    0.620 ±  0.001   ms/op
Client.existUser:existUser·p0.00      sample             0.161            ms/op
Client.existUser:existUser·p0.50      sample             0.569            ms/op
Client.existUser:existUser·p0.90      sample             0.849            ms/op
Client.existUser:existUser·p0.95      sample             0.970            ms/op
Client.existUser:existUser·p0.99      sample             1.495            ms/op
Client.existUser:existUser·p0.999     sample             4.643            ms/op
Client.existUser:existUser·p0.9999    sample            14.549            ms/op
Client.existUser:existUser·p1.00      sample            41.615            ms/op
Client.getUser                        sample   611290    1.569 ±  0.003   ms/op
Client.getUser:getUser·p0.00          sample             0.387            ms/op
Client.getUser:getUser·p0.50          sample             1.485            ms/op
Client.getUser:getUser·p0.90          sample             2.236            ms/op
Client.getUser:getUser·p0.95          sample             2.458            ms/op
Client.getUser:getUser·p0.99          sample             3.195            ms/op
Client.getUser:getUser·p0.999         sample             5.546            ms/op
Client.getUser:getUser·p0.9999        sample            20.775            ms/op
Client.getUser:getUser·p1.00          sample            29.000            ms/op
Client.listUser                       sample    90833   10.562 ±  0.036   ms/op
Client.listUser:listUser·p0.00        sample             1.585            ms/op
Client.listUser:listUser·p0.50        sample            10.437            ms/op
Client.listUser:listUser·p0.90        sample            13.009            ms/op
Client.listUser:listUser·p0.95        sample            14.434            ms/op
Client.listUser:listUser·p0.99        sample            22.807            ms/op
Client.listUser:listUser·p0.999       sample            37.104            ms/op
Client.listUser:listUser·p0.9999      sample            61.735            ms/op
Client.listUser:listUser·p1.00        sample            63.242            ms/op




优化后的：
Benchmark                               Mode      Cnt   Score     Error   Units
Client.createUser                      thrpt        3  21.220 ±  41.514  ops/ms
Client.existUser                       thrpt        3  48.470 ± 111.865  ops/ms
Client.getUser                         thrpt        3  20.601 ±  20.811  ops/ms
Client.listUser                        thrpt        3   3.032 ±   9.202  ops/ms
Client.createUser                       avgt        3   1.466 ±   0.357   ms/op
Client.existUser                        avgt        3   0.650 ±   3.448   ms/op
Client.getUser                          avgt        3   1.448 ±   0.607   ms/op
Client.listUser                         avgt        3   9.846 ±   7.369   ms/op
Client.createUser                     sample   717386   1.337 ±   0.002   ms/op
Client.createUser:createUser·p0.00    sample            0.373             ms/op
Client.createUser:createUser·p0.50    sample            1.161             ms/op
Client.createUser:createUser·p0.90    sample            1.966             ms/op
Client.createUser:createUser·p0.95    sample            2.286             ms/op
Client.createUser:createUser·p0.99    sample            3.154             ms/op
Client.createUser:createUser·p0.999   sample            5.780             ms/op
Client.createUser:createUser·p0.9999  sample           11.370             ms/op
Client.createUser:createUser·p1.00    sample           13.189             ms/op
Client.existUser                      sample  1609319   0.596 ±   0.001   ms/op
Client.existUser:existUser·p0.00      sample            0.148             ms/op
Client.existUser:existUser·p0.50      sample            0.537             ms/op
Client.existUser:existUser·p0.90      sample            0.811             ms/op
Client.existUser:existUser·p0.95      sample            0.946             ms/op
Client.existUser:existUser·p0.99      sample            1.616             ms/op
Client.existUser:existUser·p0.999     sample            4.002             ms/op
Client.existUser:existUser·p0.9999    sample           24.906             ms/op
Client.existUser:existUser·p1.00      sample           75.104             ms/op
Client.getUser                        sample   766424   1.252 ±   0.002   ms/op
Client.getUser:getUser·p0.00          sample            0.364             ms/op
Client.getUser:getUser·p0.50          sample            1.202             ms/op
Client.getUser:getUser·p0.90          sample            1.706             ms/op
Client.getUser:getUser·p0.95          sample            1.942             ms/op
Client.getUser:getUser·p0.99          sample            2.863             ms/op
Client.getUser:getUser·p0.999         sample            4.948             ms/op
Client.getUser:getUser·p0.9999        sample            9.322             ms/op
Client.getUser:getUser·p1.00          sample           16.171             ms/op
Client.listUser                       sample   106601   8.999 ±   0.022   ms/op
Client.listUser:listUser·p0.00        sample            0.941             ms/op
Client.listUser:listUser·p0.50        sample            9.290             ms/op
Client.listUser:listUser·p0.90        sample           10.781             ms/op
Client.listUser:listUser·p0.95        sample           11.551             ms/op
Client.listUser:listUser·p0.99        sample           14.270             ms/op
Client.listUser:listUser·p0.999       sample           24.025             ms/op
Client.listUser:listUser·p0.9999      sample           30.038             ms/op
Client.listUser:listUser·p1.00        sample           31.949             ms/op


看RT有明显的下降

**Pull request merged**